File: src/main/java/org/jhapy/i18n/repository/ElementLookupRepository.java
package org.jhapy.i18n.repository;

import org.jhapy.i18n.domain.ElementLookup;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;
import java.util.UUID;

public interface ElementLookupRepository extends JpaRepository<ElementLookup, UUID> {
  Optional<ElementLookup> findByElementIdOrName(UUID elementId, String name);

  Optional<ElementLookup> findByName(String name);
}

File: src/main/java/org/jhapy/i18n/repository/ElementLookupRepository.kt
package org.jhapy.i18n.repository

import org.jhapy.i18n.domain.ElementLookup
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface ElementLookupRepository : JpaRepository<ElementLookup?, UUID?> {
    fun findByElementIdOrName(elementId: UUID?, name: String?): Optional<ElementLookup>
    fun findByName(name: String?): Optional<ElementLookup>
}